### PR TITLE
User sites pages

### LINKED ISSUE #136: My Sites
Create both my and their sites pages (`/my_account/sites`, `/user_accounts/:id/sites`)

### PR DESCRIPTION
**Pull request by Allcharles**
# User Sites Pages

Created user sites pages. (`their-sites.component.ts` non-functional)

## Issues
Unable to fully implement because of the following issues:

- No route for accessing a users sites
- ShallowSites routes don't exist
- Annotations page does not properly exist
- Sites output does not contain permissions and recent audio uploads

## Closes

Addresses #136 

## Visual Changes

My Sites Page

image

Their Sites Page

image

### CODE REVIEW

**Comment by atruskie** [comment]
Add tracking issues for the problems you mentioned, cross link them here, and link to baw-server issues in any relevant place

File: src/app/component/profile/pages/sites/their-sites.component.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      (sites) =>
+        sites.map((site) => ({
+          site: {
+            label: site.name,
+            route: site.redirectPath(),
+          },
+          recentAudioUpload: "(none)",
+          permission: "FIX ME",
+          annotation: annotationsMenuItem.uri(undefined),
+        })),
```
Comment (atruskie) [inline src/app/component/profile/pages/sites/their-sites.component.ts:54]: same as projects list page - why not map this into a `Site` model?

File: src/app/models/Site.ts [hunk 2]
```diff
@@ -69,18 +69,17 @@ export class Site extends AbstractModel implements SiteInterface {
...
+      description: this.description,
     };
   }
 
-  redirectPath(project: Project): string {
+  redirectPath(project?: Project): string {
     return siteMenuItem.route.format({
-      projectId: project.id,
-      siteId: this.id
+      projectId: project?.id || this.projectIds[0] || -1,
```
Comment (atruskie) [inline src/app/models/Site.ts]: this seems dangerous - there's a possibility of a `-1` appearing in a URL? better to just return null or empty string if a valid URL is not possible

**Review by atruskie — APPROVED** [review]

File: src/app/models/Site.ts (hunk 2 above)
Comment (Allcharles) [reply src/app/models/Site.ts]: I think ill just put a check, if no project.id or this.projectIds[0], return "/BROKEN_LINK".

Comment (atruskie) [reply src/app/models/Site.ts]: No I don't think so. Then you run the risk of "/BROKEN_LINK" actually getting inserted into an href.

I'd recommend:
- throwing if you want this to be a fatal error
- or returning something falsey if you want to fail gracefully.

But returning a string with non-zero length to indicate an error in a function that returns strings is not a great idea.

**Review by atruskie — APPROVED** [review]

**Comment by Allcharles** [comment]
Baw issues:
- https://github.com/QutEcoacoustics/baw-server/issues/438
- https://github.com/QutEcoacoustics/baw-server/issues/406

**Pull request merged**